web/src/pages/api/public/media/upload-url.ts

```diff
@@ -0,0 +1,227 @@
+import { randomUUID } from "crypto";
+
+import { env } from "@/src/env.mjs";
+import { getFileExtensionFromContentType } from "@/src/features/media/server/getFileExtensionFromContentType";
+import { getMediaStorageServiceClient } from "@/src/features/media/server/getMediaStorageClient";
+import {
+  GetMediaUploadUrlQuerySchema,
+  GetMediaUploadUrlResponseSchema,
+  MediaContentType,
+} from "@/src/features/media/validation";
+import { createAuthedAPIRoute } from "@/src/features/public-api/server/createAuthedAPIRoute";
+import { withMiddlewares } from "@/src/features/public-api/server/withMiddlewares";
+import {
+  ForbiddenError,
+  InternalServerError,
+  InvalidRequestError,
+} from "@langfuse/shared";
+import { prisma } from "@langfuse/shared/src/db";
+import { instrumentAsync } from "@langfuse/shared/src/server";
+
+export default withMiddlewares({
+  POST: createAuthedAPIRoute({
+    name: "Get Media Upload URL",
+    bodySchema: GetMediaUploadUrlQuerySchema,
+    responseSchema: GetMediaUploadUrlResponseSchema,
+    successStatusCode: 201,
+    fn: async ({ body, auth }) =>
+      instrumentAsync({ name: "create-media-upload-url" }, async () => {
```

Comment:
Spans are created by Next already. I would only create one if we are interested in timings of sub-parts of this API.

Reply:
Thanks! Will remove that